genie-common-internal/src/main/java/com/netflix/genie/common/internal/dto/v4/converters/JobDirectoryManifestProtoConverter.java

```diff
@@ -0,0 +1,92 @@
+/*
+ *
+ *  Copyright 2019 Netflix, Inc.
+ *
+ *     Licensed under the Apache License, Version 2.0 (the "License");
+ *     you may not use this file except in compliance with the License.
+ *     You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *     Unless required by applicable law or agreed to in writing, software
+ *     distributed under the License is distributed on an "AS IS" BASIS,
+ *     WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *     See the License for the specific language governing permissions and
+ *     limitations under the License.
+ *
+ */
+
+package com.netflix.genie.common.internal.dto.v4.converters;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.netflix.genie.common.internal.dto.JobDirectoryManifest;
+import com.netflix.genie.common.internal.exceptions.GenieConversionException;
+import com.netflix.genie.proto.JobFileManifestMessage;
+import org.springframework.validation.annotation.Validated;
+
+import javax.validation.constraints.NotBlank;
+import java.io.IOException;
+
+/**
+ * Converts {@link JobDirectoryManifest} from/to {@link JobFileManifestMessage} in order to transport manifests
+ * over gRPC.
+ *
+ * @author mprimi
+ * @since 4.0.0
+ */
+@Validated
+public class JobDirectoryManifestProtoConverter {
+
+    private final ObjectMapper objectMapper;
+
+    /**
+     * Constructor.
+     *
+     * @param objectMapper an object mapper
+     */
+    public JobDirectoryManifestProtoConverter(final ObjectMapper objectMapper) {
+        this.objectMapper = objectMapper;
+    }
+
+    /**
+     * Construct a {@link JobFileManifestMessage} from the given {@link JobDirectoryManifest}.
+     *
+     * @param claimedJobId the id of the job this file manifest belongs to
+     * @param manifest     the manifest
+     * @return a {@link JobFileManifestMessage}
+     * @throws GenieConversionException if conversion fails
+     */
+    public JobFileManifestMessage manifestToProtoMessage(
+        @NotBlank final String claimedJobId,
+        final JobDirectoryManifest manifest
+    ) throws GenieConversionException {
+        final String manifestJsonString;
+        try {
+            // Leverage the existing manifest serialization to JSON rather than creating and using a protobuf schema.
+            manifestJsonString = objectMapper.writeValueAsString(manifest);
```

Comment:
I hear ya. We don't have a JSON schema other than the one implicitly defined by serializing this.

On one side, I'm not too worried, because we're unlikely to modify this, and if we do is to add things and we can figure it out.

On the other side, ignoring this could fire back.

Maybe we can look at some tools for JSON schema validation and add a test that validates that the manifest JSON does not change.
So if one day we do modify this class, tests will fail if the schema is not compatible.